config/src/test/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurerTests.java

```diff
@@ -0,0 +1,237 @@
+					.and()
+				.httpBasic();
+			// @formatter:on
+		}
+
+		@Override
+		protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+			// @formatter:off
+			auth
+				.inMemoryAuthentication();
+			// @formatter:on
+		}
+	}
+
+	@Test
+	public void httpBasicAllowsSettingAuthenticationEntryPoint() throws Exception {
+		this.spring.register(CustomAuthenticationEntryPointConfig.class).autowire();
+
+		this.mvc.perform(get("/"));
+
+		verify(CustomAuthenticationEntryPointConfig.ENTRY_POINT)
+				.commence(any(HttpServletRequest.class),
+						any(HttpServletResponse.class),
+						any(AuthenticationException.class));
+	}
+
+	@EnableWebSecurity
+	static class CustomAuthenticationEntryPointConfig extends WebSecurityConfigurerAdapter {
+		static AuthenticationEntryPoint ENTRY_POINT = mock(AuthenticationEntryPoint.class);
+
+		@Override
+		protected void configure(HttpSecurity http) throws Exception {
+			// @formatter:off
+			http
+				.authorizeRequests()
+					.anyRequest().authenticated()
+					.and()
+				.httpBasic()
+					.authenticationEntryPoint(ENTRY_POINT);
+			// @formatter:on
+		}
+
+		@Override
+		protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+			// @formatter:off
+			auth
+				.inMemoryAuthentication();
+			// @formatter:on
+		}
+	}
+
+	@Test
+	public void httpBasicUsesOriginalEntryPointWhenInvokedTwice() throws Exception {
+		this.spring.register(DuplicateDoesNotOverrideConfig.class).autowire();
+
+		this.mvc.perform(get("/"));
+
+		verify(DuplicateDoesNotOverrideConfig.ENTRY_POINT)
+				.commence(any(HttpServletRequest.class),
+						any(HttpServletResponse.class),
+						any(AuthenticationException.class));
+	}
+
+	@EnableWebSecurity
+	static class DuplicateDoesNotOverrideConfig extends WebSecurityConfigurerAdapter {
+		static AuthenticationEntryPoint ENTRY_POINT = mock(AuthenticationEntryPoint.class);
+
+		@Override
+		protected void configure(HttpSecurity http) throws Exception {
+			// @formatter:off
+			http
+				.authorizeRequests()
+					.anyRequest().authenticated()
+					.and()
+				.httpBasic()
+					.authenticationEntryPoint(ENTRY_POINT)
+					.and()
+				.httpBasic();
+			// @formatter:on
+		}
+
+		@Override
+		protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+			// @formatter:off
+			auth
+				.inMemoryAuthentication();
+			// @formatter:on
+		}
+	}
+
+	//SEC-3019
+	@Test
+	public void httpBasicSetsRememberMeCookieWhenConfigured() throws Exception {
+		this.spring.register(BasicUsesRememberMeConfig.class).autowire();
+
+		this.mvc.perform(get("/")
+				.with(httpBasic("user", "password"))
+				.param("remember-me", "true"))
+		.andExpect(cookie().exists("remember-me"));
+	}
+
+	@EnableWebSecurity
+	@Configuration
+	static class BasicUsesRememberMeConfig extends WebSecurityConfigurerAdapter {
+
+		@Override
+		protected void configure(HttpSecurity http) throws Exception {
+			// @formatter:off
+			http
+				.httpBasic().and()
```

Comment:
Would you please reformat this so that it follows [the team convention](https://spring.io/blog/2013/07/11/spring-security-java-config-preview-readability/)?